pipettes/tests/test_motor_pulse.cpp

```diff
@@ -0,0 +1,136 @@
+#include <stdio.h>
+
+#include <iostream>
+
+#include "catch2/catch.hpp"
+#include "motor-control/core/motor_interrupt_handler.hpp"
+#include "pipettes/tests/mock_message_queue.hpp"
+
+using namespace motor_handler;
+
+TEST_CASE("move a full rotation") {
```

Comment:
These tests aren't really testing the right layers. With this kind of step math, we really need to test the _numeric_ edge cases, not the motion-system edge cases, because it's so limited to numerics; and, whenever we start doing direction and enable in the same subsystem, those lines too. So for instance, we should have tests that exercise
- Behavior starting from `step_count=0` and going positive and negative;
- Behavior starting above 1.0 and going through 0 negative going
- Behavior starting in (1.0, 0.0) somewhere and moving through 0.0 negative going
- The same with below -1.0 and up through 0; and between 0 and -1.0 and up through 0
- Maybe all the way from > 1.0 to < -1.0 and the reverse

We can keep the overflow cases, but maybe transition to just detecting and raising an error; if we have the step counter and position tracker be the same, it can't get that big anyway.